src/Features/Core/Portable/MakeMethodSynchronous/AbstractMakeMethodSynchronousCodeFixProvider.cs

```diff
@@ -0,0 +1,110 @@
+            if (methodSymbolOpt?.MethodKind == MethodKind.Ordinary && 
+                methodSymbolOpt.Name.Length > AsyncSuffix.Length && 
+                methodSymbolOpt.Name.EndsWith(AsyncSuffix))
+            {
+                return await RenameThenRemoveAsyncTokenAsync(document, node, methodSymbolOpt, cancellationToken).ConfigureAwait(false);
+            }
+            else
+            {
+                return await RemoveAsyncTokenAsync(document, methodSymbolOpt, node, cancellationToken).ConfigureAwait(false);
+            }
+        }
+
+        private async Task<Solution> RenameThenRemoveAsyncTokenAsync(Document document, SyntaxNode node, IMethodSymbol methodSymbol, CancellationToken cancellationToken)
+        {
+            var name = methodSymbol.Name;
+            var newName = name.Substring(0, name.Length - AsyncSuffix.Length);
+            var solution = document.Project.Solution;
+            var options = solution.Workspace.Options;
+
+            // Store the path to this node.  That way we can find it post rename.
+            var syntaxPath = new SyntaxPath(node);
+
+            // Rename the method to remove the 'Async' suffix, then remove the 'async' keyword.
+            var newSolution = await Renamer.RenameSymbolAsync(solution, methodSymbol, newName, options, cancellationToken).ConfigureAwait(false);
+            var newDocument = newSolution.GetDocument(document.Id);
+            var newRoot = await newDocument.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            SyntaxNode newNode;
+            if (syntaxPath.TryResolve<SyntaxNode>(newRoot, out newNode))
+            {
+                var semanticModel = await newDocument.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var newMethod = (IMethodSymbol)semanticModel.GetDeclaredSymbol(newNode, cancellationToken);
+                return await RemoveAsyncTokenAsync(newDocument, newMethod, newNode, cancellationToken).ConfigureAwait(false);
+            }
+
+            return newSolution;
+        }
+
+        private async Task<Solution> RemoveAsyncTokenAsync(Document document, IMethodSymbol methodSymbolOpt, SyntaxNode node, CancellationToken cancellationToken)
+        {
+            var newNode = RemoveAsyncTokenAndFixReturnType(methodSymbolOpt, node);
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var newRoot = root.ReplaceNode(node, newNode);
+
+            var newDocument = document.WithSyntaxRoot(newRoot);
+            return newDocument.Project.Solution;
+        }
+
+        private class MyCodeAction : CodeAction.SolutionChangeAction
+        {
+            public MyCodeAction(
+                string title,
+                Func<CancellationToken, Task<Solution>> createChangedSolution,
+                string equivalenceKey = null) : base(title, createChangedSolution, equivalenceKey)
```

Comment:
I was hesitant about that at first... but i guess it could be ok to support that here...